amgcl/detail/qr.hpp

```diff
@@ -223,10 +223,10 @@ class QR {
             const int row_stride = (Order == row_major ? nmax : 1);
             const int col_stride = (Order == row_major ? 1 : m);
 
-            for(int i = 0, ii = 0; i < n; ++i, ii += row_stride + col_stride)
+            for(int i = 0, ii = 0; i < m; ++i, ii += row_stride + col_stride)
```

Comment:
That is correct. Also, the solver code should be able to reuse the existing decomposition code without any changes, since the decomposition interface supports both fortran and c style storage order for matrices.